src/components/tree/tree.tsx

```diff
@@ -22,12 +26,72 @@ export const spaceBetweenTrees = 100;
 export const lhsTreeId = "LEFT";
 const rhsTreeId = "RIGHT";
 
-class Tree extends React.Component {
-  constructor(props) {
+export interface TreeComponentOwnProps {
+  dispatch: AppDispatch
+  height: number
+  width: number
+}
+
+export interface TreeComponentProps extends WithTranslation, TreeComponentStateProps, TreeComponentOwnProps {}
+
+// This is duplicated from RootState, but good to be explicit about what's
+// expected here.
```

Comment:
would defining RootState in terms of this interface reduce the duplication?

Reply:
It's not trivial since thee same is different (e.g. `tree: state.tree` vs. `selectedNode: state.controls.selectedNode` vs. `colorings: state.metadata.colorings`): https://github.com/nextstrain/auspice/blob/3e23f8590764ce7863ad8e6a798ffc676008e0da/src/components/tree/index.ts#L6-L38

So it would need to be something like `type TreeComponentStateProps = PropsFromTreeState & PropsFromControlsState & PropsFromColoringsState …`.

I also think it's more logical to keep the RootState as the central source of truth.

Reply:
While looking into this, I realized that it makes more sense to put these types in a separate `types.ts` file so I've done that: 3e23f8590764ce7863ad8e6a798ffc676008e0da